services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/storage.py

```diff
@@ -15,30 +15,52 @@
 _LIVENESS_TIMEOUT: Final[timedelta] = timedelta(seconds=5)
 
 
-async def _is_storage_responsive(url: str) -> bool:
+def _get_auth(storage_auth_settings: StorageAuthSettings) -> tuple[str, str] | None:
+    if storage_auth_settings.auth_required:
+        assert storage_auth_settings.STORAGE_USERNAME  # nosec
+        assert storage_auth_settings.STORAGE_PASSWORD  # nosec
+        return (
+            storage_auth_settings.STORAGE_USERNAME,
+            storage_auth_settings.STORAGE_PASSWORD.get_secret_value(),
+        )
+    return None
+
+
+def _get_url(storage_auth_settings: StorageAuthSettings) -> str:
+    return f"{storage_auth_settings.api_base_url}/"
```

Comment:
is this ```/``` at the end really required?? can't we use URL library that takes care of these things? these are typically very annoying hacks...

Reply:
we can parse with yarl